src/RemediationRoleSaver.cpp

```diff
@@ -301,6 +318,10 @@ void ResultBasedLibraryRemediationSaver::saveToFile(const QString& filename)
     outputFile.open(QIODevice::WriteOnly);
 
     const int rc = xccdf_policy_generate_fix(policy, result, mTemplateString.toUtf8().constData(), outputFile.handle());
+    if (!outputFile.flush())
+    {
+        throw std::runtime_error("Could not write to the destination location.");
```

Comment:
This is not what I had in mind. I was just giving an example when it fails, patching just the symptom I pointed out instead of the root cause is suboptimal. IMO we should report oscap errors well via a dialog + stderr into workbench diagnostics dialog. That would solve all of these issues.

You absolutely don't want to flush the file from Qt btw, you never wrote anything to the file via the Qt interface and you opened it write only. Whether Qt empties the file or not is implementation defined.

Reply:
The problem this addresses are cases when the workbench wants to write zero-sized files to directories that are not writable. Currently, there are no `puppet` remediations, so when I want to save a remediation profile, there is no error, but the result is an empty file. Then:
* If the destination is writable, an empty file is created and stays there, whereas
* if the destination is not writable, there is no error even if no file is created. Errors happen during the `flush`, calling `outputFile.close()` silences any `flush` errors.
This inconsistency that I have discovered thank to your suggestion seemed misleading to me in to an extent that it justified this ad-hoc fix. Anyway, should I keep this, or should I remove this?